themes/default/content/resources/building-container-based-aws-lambda-functions-in-python/index.md

```diff
@@ -0,0 +1,79 @@
+
+    So we get to use this, which is great. So that's a nice convenience. So here we're using the full access managed policy ARN. And then we finally create the function specifying the image package type along with the actual image name, which is the fully qualified URL for the image that we uploaded earlier. So combine all that together, we have this function now.
+
+    And the final thing to do is to really just wire up that Lambda function we created with S3 bucket notifications. And this is also fairly simple. All we're doing here is first giving permissions for the bucket to invoke the function. And so here you can see we have this permission where we allow invoke function from S3, from that bucket that we created as the source, and then it's going to invoke this function that we just defined. And then we have to define how we actually want to invoke. And so we create this bucket notification. And in the bucket notification, again, we provide the bucket that we're actually going to invoke from along with the functions that are going to be invoked off that notification.
+
+    And so here you can see we give it this function ARN and then we give it this event. This is for object creation because obviously we care about any videos that get created, and then we filter on this MP4 suffix since we're looking for specifically video files. Just because it's easier, we actually write JPEGs into the same bucket. So we obviously want to make sure we're not infinitely recursing on ourselves. And so we're just looking at the video files. And this depends on having this permission obviously. So we have this explicit dependency we define here.
+
+    So that's how this all gets this wired up. And then we finally export the bucket name for convenience. And so I actually ran this earlier. So I ran 'pulumi up'. It provided all the resources, shoved them all into AWS. And when I run this copy of the sample file into the bucket, if I show the output, this is running Pulumi logs, I'm just tailing the logs from CloudWatch. And you can see here, this is the thumbnail or task or function that's getting invoked. And here it's copying the file from S3 into the temporary directory on the container, and then it's running FFmpeg to extract that image. So you can actually see this thumbnails is getting saved here in the sample that JPEG. So you can see here it's copying as well. So let's actually copy this locally and see that it actually worked. And if we open up that sample, there's the thumbnail. And this is a video of a waterfall. So great success. It all worked.
```

Comment:
```suggestion
    So that's how this all gets this wired up. And then we finally export the bucket name for convenience. And so I actually ran this earlier. So I ran 'pulumi up'. It provided all the resources, shoved them all into AWS. And when I run this copy of the sample file into the bucket, if I show the output, this is running Pulumi logs, I'm just tailing the logs from CloudWatch. And you can see here, this is the thumbnail or task or function that's getting invoked. And here it's copying the file from S3 into the temporary directory on the container, and then it's running FFmpeg to extract that image. So you can actually see this thumbnail is getting saved here in the sample that JPEG. So you can see here it's copying as well. So let's actually copy this locally and see that it actually worked. And if we open up that sample, there's the thumbnail. And this is a video of a waterfall. So great success. It all worked.
```